api/src/main/java/org/openmrs/module/sync/api/db/hibernate/HibernateSyncDAO.java

```diff
@@ -646,7 +646,7 @@ public void flushSession() throws DAOException {
 	 * @throws DAOException
 	 */
 	public void saveOrUpdate(Object object) throws DAOException {
-		sessionFactory.getCurrentSession().saveOrUpdate(object);
+		sessionFactory.getCurrentSession().save(object);

```

Comment:
Without this change, sync unit tests around privileges in particular failed.  Changing this just to "save" caused those unit tests to pass, and all others to continue to pass.

Reply:
What is the different between the "saveOrUpdate" and "save" method, and do we have any reason to think it matters here?  I would assume that a "save" method would fail when trying to update an existing object, but maybe not? I assume some of tests test the "update" use case.

Reply:
https://stackoverflow.com/questions/17612364/difference-between-save-and-saveorupdate-method-hibernate#:~:text=hibernate.,save

Reply:
Yeah.  I think the issue with the saveOrUpdate method with "privilege" was that it thought it needed to do an update rather than an insert since the primary key (privilege) was already populated on the object.  I think this likely works with "save" because we are working with persistent Hibernate-managed objects, not transient objects that are disconnected from the session, so Hibernate already knows that they already exist and handles them appropriately in the case of calling "save", but I can't say for sure.  I just know that this works (unit tests pass).  And we'll know pretty soon from testing on an actual OpenMRS instance whether or not it works in usage.

Reply:
Yeah, seems like if it is a problem it will fail miserably during actual testing.  Might be good to do some explicit checking in the database for duplicate objects.